frontend/src/components/SignUpPage.tsx

```diff
@@ -0,0 +1,110 @@
+import {Button, TextField} from "@mui/material";
+import React, {ChangeEvent, FormEvent, useState} from "react";
+import {useNavigate} from "react-router-dom";
+import {UserModel} from "../model/User";
+import {styled} from "@mui/material/styles";
+
+type createUserProps = {
+    createUser: (user: UserModel) => Promise<boolean>;
+}
+
+const FormContainer = styled('form')({
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '1rem',
+});
+
+export const SignUpPage = (props: createUserProps) => {
+
+    const initial: UserModel = {
+        username: "", password: ""
+    }
+    const [user, setUser] = useState<UserModel>(initial);
+    const navigate = useNavigate();
+
+    function onChange(event: ChangeEvent<HTMLInputElement>) {
+        const targetName: string = event.target.name;
+        const value: string = event.target.value;
+        setUser({
+            ...user,
+            [targetName]: value
+        })
+    }
+
+    function onSubmit(event: FormEvent<HTMLFormElement>) {
+        if (user.username && user.password) {
+            event.preventDefault();
+            props.createUser(user).then((s) => {
+                if (s) {
+                    setUser(initial);
+                    navigate('/login');
+                } else {
+                    console.log("error adding new User")
```

Comment:
do you have toastify in your project?

Reply:
yes but wont show in /login only in gallery even when i add ToastContainer.